xbmc/utils/StreamUtils.cpp

```diff
@@ -96,3 +100,64 @@ std::string StreamUtils::GetCodecName(int codecId, int profile)
 
   return codecName;
 }
+
+std::string StreamUtils::GetLayoutXYZ(uint64_t mask)
+{
+  constexpr uint64_t standardMask{
+      AV_CH_FRONT_LEFT | AV_CH_FRONT_RIGHT | AV_CH_FRONT_CENTER | AV_CH_BACK_LEFT |
+      AV_CH_BACK_RIGHT | AV_CH_FRONT_LEFT_OF_CENTER | AV_CH_FRONT_RIGHT_OF_CENTER |
+      AV_CH_BACK_CENTER | AV_CH_SIDE_LEFT | AV_CH_SIDE_RIGHT | AV_CH_STEREO_LEFT |
+      AV_CH_STEREO_RIGHT | AV_CH_WIDE_LEFT | AV_CH_WIDE_RIGHT | AV_CH_SURROUND_DIRECT_LEFT |
+      AV_CH_SURROUND_DIRECT_RIGHT | AV_CH_SIDE_SURROUND_LEFT | AV_CH_SIDE_SURROUND_RIGHT};
+  constexpr uint64_t lfeMask{AV_CH_LOW_FREQUENCY | AV_CH_LOW_FREQUENCY_2};
+  constexpr uint64_t topMask{AV_CH_TOP_CENTER | AV_CH_TOP_FRONT_LEFT | AV_CH_TOP_FRONT_CENTER |
+                             AV_CH_TOP_FRONT_RIGHT | AV_CH_TOP_BACK_LEFT | AV_CH_TOP_BACK_CENTER |
+                             AV_CH_TOP_BACK_RIGHT | AV_CH_TOP_SIDE_LEFT | AV_CH_TOP_SIDE_RIGHT |
+                             AV_CH_TOP_SURROUND_LEFT | AV_CH_TOP_SURROUND_RIGHT};
+
+  const int standardChannels{std::popcount(mask & standardMask)};
+  const int lfeChannels{std::popcount(mask & lfeMask)};
+  const int topChannels{std::popcount(mask & topMask)};
+
+  std::string layout = std::to_string(standardChannels);
+  layout.append(".");
+  layout.append(std::to_string(lfeChannels));
+  if (topChannels)
+  {
+    layout.append(".");
+    layout.append(std::to_string(topChannels));
+  }
+  return layout;
+}
+
+std::string StreamUtils::GetLayout(uint64_t mask, int channels)
+{
+  if (mask)
+    return GetLayoutXYZ(mask);
+
+  //! @todo return something or nothing in case of unknown layout?
+  std::string layout = std::to_string(channels);
+  layout.append(" ");
+  layout.append(g_localizeStrings.Get(10127)); // "channels"
+  return layout;
```

Comment:
IMO you should avoid use language localizations here
and leave the task to skins/addons etc...
otherwise the value will result impossible to be mapped

Reply:
We could mark the message po.strings as used by skins - I think it's of general interest and defining it per skin/addon would be more effort overall.

Whether to display a xx channels in case of no data or return a blank string and let the skin handle - that is the fundamental question here. I'd let skinners answer.

unknown layout: as things currently stand, when ffmpeg demuxer doesn't find a layout in the video, or (every other flow), we don't have a default layout for a channel count.

Reply:
since already exists a numeric channels
i think that mixing layouts and custom form of channels layout (with also localization) can be avoided

skins and addons have the ability to handle this
then No layout == empty string,
skins can fallback to use integer channel number and also use localization when the layout is empty

if on skin translations string dont exist a relative string for "channels" (10127) then must be add a new one on skin translations
you have not to mix kodi core translations with skin

Reply:
Added a commit to leave the infolabel blank (to be squashed later)